lib/data_update_scripts/20200805171911_clean_up_language_settings.rb

```diff
@@ -0,0 +1,19 @@
+module DataUpdateScripts
+  class CleanUpLanguageSettings
+    def run
+      User.where("(language_settings->'preferred_languages') IS null").find_each do |user|
+        language_settings = user.language_settings
+        language_settings["preferred_languages"] = []
+
+        language_settings["preferred_languages"] << "en" if language_settings["prefer_language_en"]
+        language_settings["preferred_languages"] << "es" if language_settings["prefer_language_es"]
+        language_settings["preferred_languages"] << "fr" if language_settings["prefer_language_fr"]
+        language_settings["preferred_languages"] << "it" if language_settings["prefer_language_it"]
+        language_settings["preferred_languages"] << "ja" if language_settings["prefer_language_ja"]
+        language_settings["preferred_languages"] << "pt" if language_settings["prefer_language_pt"]
```

Comment:
Just one question, is the order in the array going to have an impact on the UX?

Reply:
Good question! I had to double-check that - no, the order for the UX is defined by a constant:

`<% Languages::LIST.each do |code, name| %>`